lan867x/src/esp_eth_phy_lan867x.c

```diff
@@ -0,0 +1,346 @@
+#define ETH_PHY_IDR1_REG_ADDR (0x02)
+
+typedef union {
+    struct {
+        uint32_t model_revision : 4; /*!< Model revision number */
+        uint32_t vendor_model : 6;   /*!< Vendor model number */
+        uint32_t oui_bits_18_23 : 6; /*!< Organizationally Unique Identifier(OUI) bits 19 to 24 */
+    };
+    uint32_t val;
+} lan867x_phyidr2_reg_t;
+#define ETH_PHY_IDR2_REG_ADDR (0x03)
+
+typedef union {
+    struct {
+        uint32_t reserved1 : 14;    // Reserved
+        uint32_t rst : 1;           // PLCA Reset
+        uint32_t en : 1;            // PLCA Enable
+    };
+    uint32_t val;
+} lan867x_plca_ctrl0_reg_t;
+#define ETH_PHY_PLCA_CTRL0_REG_MMD_ADDR (0xCA01)
+
+typedef union {
+    struct {
+        uint8_t id;     // PLCA ID
+        uint8_t ncnt;   // Node count
+    };
+    uint32_t val;
+} lan867x_plca_ctrl1_reg_t;
+#define ETH_PHY_PLCA_CTRL1_REG_MMD_ADDR (0xCA02)
+
+typedef union {
+    struct {
+        uint8_t maxbc;  // Maximum burst count
+        uint8_t btmr;   // Burst timer
+    };
+    uint32_t val;
+} lan867x_plca_burst_reg_t;
+#define ETH_PHY_PLCA_BURST_REG_MMD_ADDR (0xCA05)
+
+typedef union {
+    struct {
+        uint8_t entries[2];
+    };
+    uint32_t val;
+} lan867x_plca_multiple_id_reg_t;
+#define ETH_PHY_PLCA_MULTID_BASE_MMD_ADDR (0x0030)
+
+typedef struct {
+    phy_802_3_t phy_802_3;
+} phy_lan867x_t;
+
+#define MISC_REGISTERS_DEVICE   0x1f
+
+/***********Custom functions implementations***********/
+esp_err_t esp_eth_phy_lan867x_read_oui(phy_802_3_t *phy_802_3, uint32_t *oui)
+{
+    esp_err_t ret = ESP_OK;
+    lan867x_phyidr1_reg_t id1;
+    lan867x_phyidr2_reg_t id2;
+    esp_eth_mediator_t *eth = phy_802_3->eth;
+
+    ESP_GOTO_ON_FALSE(oui != NULL, ESP_ERR_INVALID_ARG, err, TAG, "oui can't be null");
+
+    ESP_GOTO_ON_ERROR(eth->phy_reg_read(eth, phy_802_3->addr, ETH_PHY_IDR1_REG_ADDR, &(id1.val)), err, TAG, "read ID1 failed");
+    ESP_GOTO_ON_ERROR(eth->phy_reg_read(eth, phy_802_3->addr, ETH_PHY_IDR2_REG_ADDR, &(id2.val)), err, TAG, "read ID2 failed");
+
+    *oui = (id2.oui_bits_18_23 << 18) + ((id1.oui_bits_10_17 << 10) + (id1.oui_bits_2_9 << 2));
+    return ESP_OK;
+err:
+    return ret;
+}
+
+static esp_err_t lan867x_update_link_duplex_speed(phy_lan867x_t *lan867x)
+{
+    esp_err_t ret = ESP_OK;
+    esp_eth_mediator_t *eth = lan867x->phy_802_3.eth;
+    uint32_t addr = lan867x->phy_802_3.addr;
+    bmcr_reg_t bmcr;
+    bmsr_reg_t bmsr;
+    ESP_GOTO_ON_ERROR(eth->phy_reg_read(eth, addr, ETH_PHY_BMCR_REG_ADDR, &(bmcr.val)), err, TAG, "read BMCR failed");
+    ESP_GOTO_ON_ERROR(eth->phy_reg_read(eth, addr, ETH_PHY_BMSR_REG_ADDR, &(bmsr.val)), err, TAG, "read BMSR failed");
+    eth_speed_t speed = bmcr.speed_select ? ETH_SPEED_100M : ETH_SPEED_10M;
+    eth_duplex_t duplex = bmcr.duplex_mode  ? ETH_DUPLEX_FULL : ETH_DUPLEX_HALF;
+    eth_link_t link = bmsr.link_status ? ETH_LINK_UP : ETH_LINK_DOWN;
+    // This will be ran exactly once, when everything is setting up
```

Comment:
Try to run start/stop test with LAN8670.

Reply:
It will most probably not work. I observed similar behavior in my other driver. The fix needs to be firstly introduced in ESP-IDF. Therefore, let's keep it here as is for now to move forward. However, please create JIRA to track the issue.

Reply:
@bogdankolendovskyy please create a JIRA and provide reference number.

Reply:
Works without any issues on current gitlab master.